### PR TITLE
node: allow to override enable encapsulation on a per-node basis

### PR DESCRIPTION
**Pull request by giorio94**
Introduce the possibility to configure whether we need to enforce encapsulation-related routes and alike on a per-node basis, to allow further flexibility, useful for instance in testing environments.

### CODE REVIEW

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -1010,7 +1011,7 @@ func (n *linuxNodeHandler) nodeUpdate(oldNode, newNode *nodeTypes.Node, firstAdd
 		return errs
 	}
 
-	if n.nodeConfig.EnableAutoDirectRouting {
+	if n.nodeConfig.EnableAutoDirectRouting && !n.enableEncapsulation(newNode) {
```
Comment (YutaroHayakawa) [inline pkg/datapath/linux/node.go:1014]: I think we don't need this because in the datapath, tunnel redirection happens before kernel routing.

**Review by YutaroHayakawa — APPROVED** [review]
LGTM 👍

**Comment by giorio94** [comment]
Conformance Ginkgo hit https://github.com/cilium/cilium/issues/29236 and https://github.com/cilium/cilium/issues/29108. Succeeded after rerun.

**Pull request merged**